### PR TITLE
run e2e test locally against openshift-4 cluster

### PR DESCRIPTION
**Pull request by nurali-techie**
Fix: https://jira.coreos.com/browse/CRT-342

### CODE REVIEW

File: make/docker.mk [hunk 1]
```diff
@@ -1,4 +1,4 @@
 .PHONY: docker-image
 ## Build the docker image locally that can be deployed (only contains bare operator)
 docker-image: build
-	$(Q)docker build -f build/Dockerfile -t docker.io/${GO_PACKAGE_ORG_NAME}/${GO_PACKAGE_REPO_NAME}:${GIT_COMMIT_ID_SHORT} .
+	$(Q)docker build -f build/Dockerfile -t ${IMAGE_NAME} .
```
[the remark this answers is not in the document]
Comment (nurali-techie) [reply make/docker.mk]: I guess this is what you asking for https://github.com/codeready-toolchain/toolchain-operator/blob/master/make/dev.mk#L2 .. it's already there.

File: make/test.mk [hunk 2]
```diff
@@ -102,15 +107,20 @@ e2e-setup: is-minishift
...
 	$(shell echo "oc login -u system:admin")
 	$(eval IMAGE_NAME := docker.io/${GO_PACKAGE_ORG_NAME}/${GO_PACKAGE_REPO_NAME}:${GIT_COMMIT_ID_SHORT})
-	$(shell echo "make docker-image")
-else
+	$(MAKE) docker-image IMAGE_NAME=${IMAGE_NAME}
+else ifneq ($(IS_OS_CI),)
 	$(eval IMAGE_NAME := registry.svc.ci.openshift.org/${OPENSHIFT_BUILD_NAMESPACE}/stable:toolchain-operator)
+else
+	# For OpenShift-4
+	$(eval IMAGE_NAME := quay.io/${QUAY_NAMESPACE}/${GO_PACKAGE_REPO_NAME}:${DATE_SUFFIX})
```
[the remark this answers is not in the document]
Comment (nurali-techie) [reply make/test.mk]: QUAY_NAMESPACE is not set to anything.  One has to set `QUAY_NAMESPACE` before running the test against OS-4 cluster.  This is similar to done here https://github.com/codeready-toolchain/toolchain-e2e/blob/master/make/test.mk#L239 .. Is there any other expectation ??

[the remark this answers is not in the document]
Comment (MatousJobanek) [reply make/test.mk]: no. I was blind, I overlooked that - sorry for confusion and thaks for explanation

File: make/docker.mk (hunk 1 above)
[the remark this answers is not in the document]
Comment (MatousJobanek) [reply make/docker.mk]: I would expect that the variable would be rather in this file than in `dev.make`, but anyway the `IMAGE_NAME` is set to `IMAGE_NAME?=toolchain-operator` there and I don't think that it's the correct image name. 
How about using something like this:
```
QUAY_NAMESPACE ?= ${GO_PACKAGE_ORG_NAME}
TARGET_REGISTRY := quay.io
IMAGE_NAME ?= ${TARGET_REGISTRY}/${QUAY_NAMESPACE}/${GO_PACKAGE_REPO_NAME}:${GIT_COMMIT_ID_SHORT}
```

[the remark this answers is not in the document]
Comment (nurali-techie) [reply make/docker.mk]: When this target used directly, IMAGE_NAME=toolchain-operator is not bad.  Also, it's not for direct use.  It's used by the target which `build-image` and they set IMAGE_NAME appropriately.  So I agree there are few betterment can be done but not useful and can be done when we actually want them to use with particular use-case.

**Pull request merged**